src/main/java/seedu/rc4hdb/ui/history/CommandHistory.java

```diff
@@ -0,0 +1,95 @@
+package seedu.rc4hdb.ui.history;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+import javafx.scene.input.KeyCode;
+
+/**
+ * The component that is responsible for logging and keeping track of user command inputs.
+ * There is no limitation to the number of commands it can keep track. The stack is cleared upon
+ * exiting the system.
+ */
+public class CommandHistory {
```

Comment:
Should we create `CommandHistory` as an abstract class, with `UpCommandHistory` and `DownCommandHistory` as the concrete classes? This way you can use polymorphism to replace the if-else checks, and also test each class separately.